api/context/src/main/java/io/opentelemetry/context/propagation/TextMapPropagatorBuilder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context.propagation;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+class TextMapPropagatorBuilder {
+  final List<TextMapPropagator> delegates;
+  Collection<String> allFields;
+  boolean stopExtractAfterFirst = false;
+  boolean iterateBackwards = false;
+
+  TextMapPropagatorBuilder(Iterable<TextMapPropagator> delegates) {
+    this.delegates = new ArrayList<>();
+    for (TextMapPropagator propagator : delegates) {
+      requireNonNull(propagator, "propagator");
+      this.delegates.add(propagator);
+    }
+  }
+
+  public TextMapPropagatorBuilder stopExtractAfterFirst() {
+    this.stopExtractAfterFirst = true;
+    return this;
+  }
+
+  public TextMapPropagatorBuilder iterateBackwards() {
```

Comment:
Do we need / use iterateBackwards?

Reply:
Good call out. Because the original impl was in the external (fake contrib) module, it's hard to get a sense of how widely it might be used...so I am inclined to remove it.  Curious what other folks think. 

I mean, if you want reverse iteration order, reverse the list first?  🤓